Makefile

```diff
@@ -1,21 +1,91 @@
-RETOOL=$(CURDIR)/_tools/bin/retool
-PATH := ${PWD}/bin:${PWD}/ENV/bin:${PATH}
+SHELL := /bin/bash -o pipefail
+UNAME_OS := $(shell uname -s)
+UNAME_ARCH := $(shell uname -m)
+
+TMP_BASE := .tmp
+TMP := $(TMP_BASE)/$(UNAME_OS)/$(UNAME_ARCH)
+TMP_BIN = $(TMP)/bin
+
+DEP_VERSION := 0.5.0
+DEP := $(TMP_BIN)/dep
+ifeq ($(UNAME_OS),Darwin)
+DEP_OS := darwin
+endif
+ifeq ($(UNAME_OS),Linux)
+DEP_OS = linux
+endif
+ifeq ($(UNAME_ARCH),x86_64)
+DEP_ARCH := amd64
+endif
+$(DEP):
+	@mkdir -p "$(TMP_BIN)"
+	curl -sSL "https://github.com/golang/dep/releases/download/v$(DEP_VERSION)/dep-$(DEP_OS)-$(DEP_ARCH)" -o "$(DEP)"
+	@chmod +x "$(DEP)"
+
+PROTOC_VERSION := 3.7.0
```

Comment:
For clarity I would move all the `VERSION` constants to the top. I realize this is bike shedding but I think it would make it easier to understand what this makefile installs and at what version than having to search throughout the file.

Reply:
Happy to do it, I would disagree personally because this groups things nicely - if I care about `protoc`, everything is here. But can do this, no problem.

Reply:
Done